### ISSUE TITLE
Restrict package restore to primary host only

### ISSUE THREAD

**Issue by fabiocav** [Azure/azure-webjobs-sdk-script#617]
Packages should only be restored by the primary host. Secondary hosts should reload the function using the restore artifacts created by the primary host once the restore operation is completed.

**Comment by mathewc** [comment]
@fabiocav Is this already done?

**Comment by fabiocav** [comment]
No. We took care of some of the foundational pieces with the primary host identification and logging from primary, but this has not been done yet. This would likely need to be bundled with the work to cache compilation results and perform compilation on the primary host only.
